File: apps/web/repositories/mock/mock-stock-repository.ts
import type { StockRepository } from "@/repositories/stock-repository";
import { getMockPriceHistory, getMockStockDetail, getMockStockList } from "@/repositories/mock/mock-source";
import type { PriceBar, StockDetail, StockOverview, Timeframe } from "@/types/stock";

export class MockStockRepository implements StockRepository {
  async listStocks(): Promise<StockOverview[]> {
    return getMockStockList();
  }

  async getFeaturedStocks(): Promise<StockOverview[]> {
    return getMockStockList().slice(0, 4);
  }

  async getStockDetail(ticker: string): Promise<StockDetail | null> {
    return getMockStockDetail(ticker);
  }

  async getPriceHistory(ticker: string, timeframe: Timeframe): Promise<PriceBar[]> {
    return getMockPriceHistory(ticker, timeframe);
  }
}

File: apps/api/app/repositories/mock_stock_repository.py
from app.repositories.mock_data import STOCK_OVERVIEW_MAP, STOCK_SEEDS, build_price_history
from app.repositories.stock_repository import StockRepository
from app.schemas.stock import PriceBar, StockDetail, StockOverview, Timeframe


class MockStockRepository(StockRepository):
    def list_stocks(self) -> list[StockOverview]:
        return [STOCK_OVERVIEW_MAP[seed["ticker"]] for seed in STOCK_SEEDS]

    def get_stock_detail(self, ticker: str) -> StockDetail | None:
        stock = STOCK_OVERVIEW_MAP.get(ticker.upper())
        return StockDetail.model_validate(stock.model_dump()) if stock else None

    def get_price_history(self, ticker: str, timeframe: Timeframe) -> list[PriceBar]:
        return build_price_history(ticker, timeframe)

